chalice/chalicelib/request_handler.py

```diff
@@ -0,0 +1,128 @@
+import hashlib
+import json
+import tempfile
+import boto3
+
+from enum import Enum
+from typing import List
+from cloud_blobstore import BlobNotFoundError
+from chalicelib.config import REQUEST_STATUS_BUCKET_NAME, JSON_SUFFIX, \
+    MERGED_MTX_BUCKET_NAME, REQUEST_TEMPLATE, TEMP_DIR, s3_blob_store
+
+
+class RequestStatus(Enum):
+    INITIALIZED = "INITIALIZED"
+    RUNNING = "RUNNING"
+    DONE = "DONE"
+    ABORT = "ABORT"
+
+
+class RequestHandler:
+    s3 = boto3.resource("s3")
+
+    @staticmethod
+    def generate_request_id(bundle_uuids: List[str]) -> str:
+        """
+        Generate a request id based on a list of bundle uuids.
+        :param bundle_uuids: A list of bundle uuids.
+        :return: Request id.
+        """
+        bundle_uuids.sort()
+        m = hashlib.sha256()
```

Comment:
If you want to use a checksum to uniquely identify a run to avoid doing it again, you will need bundle fqids.  uuids is insufficient, as you can get a different version on a subsequent run.

Reply:
@ttung This is an important point. I am going to add a ticket for this, but may be added in a subsequent PR